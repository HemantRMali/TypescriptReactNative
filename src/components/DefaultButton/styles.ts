import {StyleSheet} from 'react-native';
import commonStyles from '../../assets/styles/styles';
import colors from '../../assets/colors/colors';
export default StyleSheet.create({
  ...commonStyles,
  commonButtonStyle: {
    padding: 15,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
    margin: 20,
  },

  buttonTitle: {
    fontSize: 20,
    color: colors.white,
  },
});
