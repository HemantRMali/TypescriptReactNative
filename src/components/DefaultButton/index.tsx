import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import {colors} from '../../assets/colors';
import {ButtonVariationsEnum} from '../../enums';
import styles from './styles';
interface defaultButtonProps {
  onPress?(): void;
  title: string;
  variationType?: string;
}

const DefaultButton = (props: defaultButtonProps) => {
  let buttonStyle;
  let buttonTitleColor = colors.steelblue;
  switch (props.variationType) {
    case ButtonVariationsEnum.primary:
      buttonStyle = styles.primary;
      buttonTitleColor = colors.white;
      break;
    case ButtonVariationsEnum.secondary:
      buttonStyle = styles.secondary;
      break;
    case ButtonVariationsEnum.third:
      buttonStyle = styles.third;
      break;
    default:
      buttonStyle = styles.primary;
      break;
  }
  return (
    <TouchableOpacity onPress={props.onPress}>
      <View style={buttonStyle}>
        <Text style={[styles.buttonTitle, {color: buttonTitleColor}]}>
          {props.title}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

DefaultButton.defaultProps = {
  variationType: ButtonVariationsEnum.primary,
};
export default DefaultButton;
