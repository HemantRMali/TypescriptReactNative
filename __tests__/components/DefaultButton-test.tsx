// __tests__/DefaultButton-test.js
import React from 'react';
import DefaultButton from '../../src/components/DefaultButton/DefaultButton';

import {render, fireEvent} from '@testing-library/react-native';

describe('DefaultButton', () => {
  it('renders properly', () => {
    const onPressHandler = jest.fn();
    const {toJSON, getByTestId} = render(
      <DefaultButton
        testId={'default-button'}
        title="Test Title"
        onPress={onPressHandler}
      />,
    );

    fireEvent.press(getByTestId('default-button'));
    expect(toJSON()).toMatchSnapshot();
  });
});
