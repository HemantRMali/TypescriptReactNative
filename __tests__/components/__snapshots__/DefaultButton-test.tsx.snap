// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultButton renders properly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="default-button"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
          "margin": 20,
          "padding": 15,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
          },
          undefined,
        ]
      }
    >
      Test Title
    </Text>
  </View>
</View>
`;
